integration-tests/tests/cases/mpc.rs

```diff
@@ -0,0 +1,230 @@
+
+    let mut data = BTreeMap::new();
+    for node in &network.nodes {
+        let id = node.me;
+        match &*node.state.test_key_info_watcher.borrow() {
+            Some(key) => {
+                data.insert(id, key.clone());
+            }
+            None => {
+                panic!("No key generated for node {id:?}")
+            }
+        }
+    }
+
+    if WRITE_OUTPUT_TO_FILES {
+        let abs_path = std::env::current_dir().unwrap().join(KEY_SHARE_FILE);
+        tracing::info!("Writing output to {}", abs_path.display());
+        let mut file = fs::File::create(KEY_SHARE_FILE).unwrap();
+        serde_json::to_writer_pretty(&mut file, &data).unwrap();
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_basic_generate_triples() {
+    let network = MpcFixtureBuilder::default()
+        .with_preshared_key()
+        .with_min_presignatures_stockpile(0)
+        .with_max_presignatures_stockpile(0)
+        .build()
+        .await;
+
+    tokio::time::timeout(Duration::from_secs(60), network.wait_for_triples(1))
+        .await
+        .expect("should have enough triples eventually");
+
+    let mut conn = network.redis_container.pool().get().await.unwrap();
+    let mut data = BTreeMap::new();
+    for node in &network.nodes {
+        let mut nodes_shares = BTreeMap::new();
+        for peer in &network.nodes {
+            let triple_ids = node.triple_storage.fetch_owned(peer.me).await;
+            let mut peer_triples = Vec::with_capacity(triple_ids.len());
+            for triple_id in triple_ids {
+                let t = conn
+                    .hget::<&str, u64, Triple>(node.triple_storage.triple_key(), triple_id)
+                    .await;
+                if let Ok(t) = t {
+                    peer_triples.push(t);
+                } else {
+                    tracing::error!("missing triple in redis {triple_id}");
+                }
+            }
+            nodes_shares.insert(peer.me, peer_triples);
+        }
+        data.insert(node.me, nodes_shares);
+    }
+
+    if WRITE_OUTPUT_TO_FILES {
+        let abs_path = std::env::current_dir().unwrap().join(TRIPLES_FILE);
+        tracing::info!("Writing output to {}", abs_path.display());
+        let mut file = fs::File::create(TRIPLES_FILE).unwrap();
+        serde_json::to_writer_pretty(&mut file, &data).unwrap();
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_basic_generate_presignature() {
+    let network = MpcFixtureBuilder::default()
+        .with_preshared_key()
+        .with_preshared_triples()
+        .with_min_triples_stockpile(0)
+        .with_max_triples_stockpile(0)
+        .build()
+        .await;
+
+    tokio::time::timeout(Duration::from_secs(10), network.wait_for_presignatures(1))
+        .await
+        .expect("should have enough presignatures eventually");
+
+    let mut conn = network.redis_container.pool().get().await.unwrap();
+    let mut data = BTreeMap::new();
```

Comment:
was there something we needed to do with `data` after collecting the shares?

Reply:
No, we just collect it to then write it out to a file if `WRITE_OUTPUT_TO_FILES` is true

It might be better to put the entire thing inside the `if WRITE_OUTPUT_TO_FILES { }` block to make that clearer. I will do so